doc-content/enterprise-only/pages/building-custom-dashboard-widgets-creating-process-heatmap-components-proc.adoc

```diff
@@ -0,0 +1,98 @@
+* You have created a project in {CENTRAL} and it contains at least one business process asset.
+* A project with a process definition has been deployed in {CENTRAL}.
+* Sample process instances are created.
+
+.Procedure
+
+. Start {CENTRAL}, go to *Menu → Design → Pages*.
+. In the *Pages* panel, click *New*.
+. In the *New Page* dialog box, enter a value in the *Name* field and select the required style.
+. Click *OK*. The new page opens in the *Page Editor*.
++
+You can see that *Heatmaps* component is available under *Components* panel.
+. Click *Save*.
+. To create a new {KIE_SERVER} dataset, complete the following steps:
+... In {CENTRAL}, go to *Admin → Data Sets*.
++
+The *Data Set Explorer* page opens.
+... Click *New Data Set* and select the *Execution Server* provider type.
++
+The *Data Set Creation Wizard* page opens.
+... Enter the name of the dataset.
+... Select the server configuration. It is available if you have deployed the project.
+... Select *CUSTOM* query target from the list.
+... In the *Query* field, enter the following custom SQL query.
++
+[source, SQL]
+----
+select
+   pil.externalId,
+   pil.processId,
+    nid,
+    nodetype,
+    nodename,
+    count(nid) as total_hits,
+    avg(execution_time) as averageExecutionTime,
+    min(execution_time) as minExecutionTime,
+    max(execution_time) as maxExecutionTime
+from(
+        select
+                max(log_date) as lastLog,
+                processinstanceid as piid,
+                nodeinstanceid as niid,
+                nodeid as nid,
+                nodetype,
+                nodename,
+                DATEDIFF(SECOND, min(log_date), max(log_date)) as execution_time
+        from
+                NodeInstanceLog
+        group by
+              processinstanceid,
+               nodeinstanceid,
+              nid
+       order by lastLog
+)
+inner join
+     ProcessInstanceLog pil on pil.processInstanceId = piid
+group by
+     pil.externalId,
+     nid,
+     nodename
+----
+
+... Complete the *Data Set Creation Wizard* and click *Test*.
+... Click *Save*.
+. Return to *Menu → Design → Pages*.
+. In the *Components* panel, expand the *Heatmaps* components and drag the *Process Heatmap* component type to the canvas.
+. In the *Displayer editor* wizard, click *Data* tab and select the newly created {KIE_SERVER} dataset.
+. In the *Data* tab, select *NID* and *AVERAGEEXECUTIONTIME* from the *Columns* field.
+. Click *Display* tab and edit the following values as per your requirements:
+* *Chart*
+* *Margins*
+* *Filter*
+* *Refresh*
+* *Columns*
+. Click *Component Editor* tab and input the following values into the *Component Properties* field:
+* *Server Template*
+* *Container ID*
+* *Process Definition ID*
++
+[NOTE]
+====
+All the fields are mandatory.
+====
+. Click *+OK*.
++
+.Example process heatmap components
+image::pages/process-heatmap-component.png[]
+
+You can see the heatmap components on the process diagram.
```

Comment:
better use `You can see the heats on the process diagram.`

Reply:
Done